---
title: "R markdown notebook of bioinformatics pipeline used in GMB304 Methods in functional genetics and development course - Workflow 2 tips"
output:
  html_notebook:
    fig_caption: yes
    toc: true
    code_folding: show
---

```{r}
### This part of the work package 3 requires you to modify and write code based on what you have learned. If there is any code in following chunks of R markdown code those are meant to be either tips or particularly hard parts of code of which learning is outside of the context of this course.
```

# 1.	Search for proliferative progenitors of 5-HT cells
a)	Use Nkx2-2 gene as marker for proliferative cells, projecting its expression to UMAP
```{r}
# Pick and modify code from previous steps
```

```{r}
# Seurat library also allows one to calculate geneset activity scores, or module scores as they call it. Basically it is a calculation that approximates how active certain set of genes is in each cells. And this number then can be projected into UMAP again. Below we calculate cell cycle related scores, please note that genes which are linked to cell-cycle stage and progression are defined within Seurat as ready sets. Following plots show scores over UMAP for various stages of cell cycle as well as best guess of the Seurat functions that in which stage each cell has been at the time of fixation
```

b) Calculating cell cycle score for cells as projecting it to UMAP (Question 9)
```{r}
# A list of cell cycle markers, from Tirosh et al, 2015, is loaded with Seurat into cc.genes object.  We can
# segregate this list into markers of G2/M phase and markers of S phase
s.genes <- str_to_title(cc.genes$s.genes)
g2m.genes <- str_to_title(cc.genes$g2m.genes)

# Use CellCycleScoring() function from Seurat to calculate scores into you dataobject
# Plot S score and G2M score as they would be gene expression values
```
# 2 Fetch WikiPathways data
```{r Fetch WikiPathways data}
#
# This step is bit tricky and given ready here
wp.mm.gmt <- rWikiPathways::downloadPathwayArchive(organism="Mus musculus", format = "gmt")
wp2gene <- readPathwayGMT(wp.mm.gmt)
wpid2gene <- wp2gene %>% dplyr::select(wpid,gene) #TERM2GENE
wpid2name <- wp2gene %>% dplyr::select(wpid,name) #TERM2NAME
```

a) Calculate and plot ORA for WikiPathways (Question 10)
```{r Calculate ORA}
# Calculate equivalent steps for WikiPathways ORA as previous ORA analyses
```

b) Explore WikiPathways for further insight
```{r}
# Wikipathways can be further studied on browser
```

# 3	Find Nkx2-2 positive progenitor and Fev positive precursor cell groups. Do differential expression comparison between these two groups to characterize main differences in their transcriptomes.
```{r}
# Find Nkx2-2 positive progenitor group with FeaturePlot
# Find Fev positive precursor cell group with FeaturePlot
# Run FindMarkers() between these groups
# What do you find? Would it make sense to look some ORA enrichment among genes specific for either one?
```
