---
title: "R markdown notebook of bioinformatics pipeline used in GMB304 Methods in functional genetics and development course - Workflow 2"
output:
  html_notebook:
    fig_caption: yes
    toc: true
    code_folding: show
---



# 1.	Search for proliferative progenitors of 5-HT cells
a)	Using Nkx2-2 gene as marker for proliferative cells, projecting its expression to UMAP
```{r}
FeaturePlot(E13vR.filt, features = c("Nkx2-2"), cols = c("lightgrey", "darkred"))
```

```{r}
# Seurat library also allows one to calculate geneset activity scores, or module scores as they call it. Basically it is a calculation that approximates how active certain set of genes is in each cells. And this number then can be projected into UMAP again. Below we calculate cell cycle related scores, please note that genes which are linked to cell-cycle stage and progression are defined within Seurat as ready sets. Following plots show scores over UMAP for various stages of cell cycle as well as best guess of the Seurat functions that in which stage each cell has been at the time of fixation
```

b) Calculating cell cycle score for cells as projecting it to UMAP (Question 9)
```{r}
# A list of cell cycle markers, from Tirosh et al, 2015, is loaded with Seurat.  We can
# segregate this list into markers of G2/M phase and markers of S phase
s.genes <- str_to_title(cc.genes$s.genes)
g2m.genes <- str_to_title(cc.genes$g2m.genes)
E13vR.filt <- CellCycleScoring(E13vR.filt, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
FeaturePlot(E13vR.filt, features = c("S.Score"), cols = c("lightgrey", "darkred"))
FeaturePlot(E13vR.filt, features = "G2M.Score", cols = c("lightgrey", "darkred"))
DimPlot(E13vR.filt)
```

# 2 Find Nkx2-2 positive progenitor and Fev positive precursor cell groups. Do differential expression comparison between these two groups to characterize main differences in their transcriptomes.
```{r}
# Find Nkx2-2 positive progenitor group with FeaturePlot
# Find Fev postivie precursor cell group with FeaturePlot
# Run FindMarkers between these groups
# What do you find? Would it make sense to look some ORA enrichment among genes specific for either one?
```

# 3 Find GABAergic neurons in the dataset 
```{r}
# Remember that GABAergic neurons are marked by Gad1 and Gad2 genes
# Draw FeaturePlot for these genes to identify main clusters of GABAergic neurons
```

# 4 Find other genes marking GABAergic neurons when compared to the rest of the cells
```{r}
# Perform DEG search between GABAergic neurons and the rest of the cells
# What are the top markers of GABAergic neurons in this dataset?
```

# Extra fetch WikiPathways data
```{r Fetch WikiPathways data}
# This exercise requires rWikiPathways package and it fethes data from WikiPathways database, however this haven't been working properly any more. So I have downloaded the data manually on 10th Dec 2024 and placed it to my working directory. Below is the code that would fetch the data automatically if it worked.
wp.mm.gmt <- rWikiPathways::downloadPathwayArchive(organism="Mus musculus", format = "gmt", date="20241210",destpath="~/my-work/")
wp2gene <- readPathwayGMT(paste("~/my-work/",wp.mm.gmt,sep=""))

# Or you can read the data from local gmt file that have been downloaded earlier
wp2gene <- readPathwayGMT("~/GMB304/wikipathways-20201110-gmt-Mus_musculus.gmt")
wpid2gene <- wp2gene %>% dplyr::select(wpid,gene) #TERM2GENE
wpid2name <- wp2gene %>% dplyr::select(wpid,name) #TERM2NAME
```

a) Calculate and plot ORA for WikiPathways (Question 10)
```{r Calculate ORA}
Wiki.ORA <- enricher(list.of.genes.up.entrez$ENTREZID, TERM2GENE = wpid2gene, TERM2NAME = wpid2name)
ewp <- DOSE::setReadable(Wiki.ORA, org.Mm.eg.db, keyType = "ENTREZID")
dotplot(Wiki.ORA, showCategory = 20)
head(Wiki.ORA)
```

b) Explore WikiPathways for further insight
```{r}
ewp.wpids <- ewp$ID
url <- getPathwayInfo("WP1266")$url
browseURL(url)
```